### ISSUE TITLE
Need a way to exit Quick Screens

### ISSUE THREAD

**Issue by cpuguru07** [bitfocus/companion-module-logos-proclaim#3]
When using a Quick Screen (Blank, Logo, etc.), we need a way to return to the Last Shown slide.
The Quick Screens in Proclaim reflect this, but it does not seem their API has exposed this feature.

I can't find a way to work-around this issue given the current limitations of this module/the Proclaim API, so I posted a feature request in their feedback forum:
https://feedback.logos.com/boards/proclaim/posts/api-improvement-quick-screens-last-shown

If anyone has a work-around in the meantime, that would be appreciated.

**Comment by hussra** [comment]
Yes, absolutely, one of a number of gaps in the published API at present, which I've previously noted on the Faithlife forums but will also add to the feedback board - thanks for posting that.

**Comment by cpuguru07** [comment]
@hussra I looked for other posts on their feedback board regarding the API, but couldn't find them. If you have them and want to link me to them, I'll upvote/comment on them as well.

**Comment by hussra** [comment]
No, it was at https://faithlife.com/posts/5621548 that I had commented, rather than their feedback board.

**Comment by soulgriever** [comment]
While investigating this, I found that Proclaim exposes an undocumented API endpoint:

/appCommand/perform?appCommandName=ShowText

In testing, this command exits the "Show No Text Quick Screen".

Example:
curl http://PROCLAIM_IP:PORT/appCommand/perform?appCommandName=ShowText

It does not appear to exit all Quick Screens (Blank, Logo, etc.), so it likely does not fully resolve this issue.

However, I submitted a PR to add support for this undocumented command to the module:

PR: [Add ShowText command to exit Quick Screens](https://github.com/bitfocus/companion-module-logos-proclaim/pull/22)

**Comment by hussra** [comment]
@soulgriever - great discovery! I wonder if there's any other undocumented endpoints that could be useful here.

**Comment by hussra** [comment]
http://PROCLAIM_IP:PORT/appCommand/perform?appCommandName=ShowLastSlide is what we are looking for.

**Comment by hussra** [comment]
Huge thanks @soulgriever - this sent me down a rabbit hole and as well as the ShowLastSlide app command I have additionally located the ?appComandName=ShowCustomQuickScreen&index=x endpoint for showing custom quick screens 1 and 2

Will get these all added to the module.

**Comment by soulgriever** [comment]
I didn't fully cross-reference, but I would imagine any command in the MIDI Input section can be added. Glad I could help!

**Comment by hussra** [comment]
I think the only gaps remaining might be one or two of the quick screens. (I confess we don't use quick screens.)

Still several other places I wish the API offered a bit more - on screen Bible, pager, and some status feedbacks other than whether you're on air.

**Comment by hussra** [comment]
v1.2.0 adds actions and presets for:

- Show Text (after using No Text)
- Show Custom Quick Screen 1/2
- Show Last Slide (after using the Blank/Logo/Custom 1/Custom 2 Quick Screens)

**Comment by hussra** [comment]
v1.2.2 of the module got released by BitFocus :-)
